pyomo/contrib/mindtpy/single_tree.py

```diff
@@ -0,0 +1,253 @@
+from __future__ import division
+
+
+from pyomo.core import Constraint, Expression, Objective, minimize, value, Var
+from pyomo.opt import TerminationCondition as tc
+from pyomo.contrib.mindtpy.nlp_solve import (solve_NLP_subproblem,
+                                             handle_NLP_subproblem_optimal, handle_NLP_subproblem_infeasible,
+                                             handle_NLP_subproblem_other_termination, solve_NLP_feas)
+from pyomo.contrib.gdpopt.util import copy_var_list_values, identify_variables
+from math import copysign
+from pyomo.environ import *
+from pyomo.core.expr import current as EXPR
+from math import fabs
+from pyomo.repn import generate_standard_repn
+import logging
+from pyomo.common.dependencies import attempt_import
+import cplex
+from cplex.callbacks import LazyConstraintCallback
+
+
+class LazyOACallback_cplex(LazyConstraintCallback):
+    """Inherent class in Cplex to call Lazy callback."""
+
+    def copy_lazy_var_list_values(self, opt, from_list, to_list, config,
+                                  skip_stale=False, skip_fixed=True,
+                                  ignore_integrality=False):
+        """Copy variable values from one list to another.
+
+        Rounds to Binary/Integer if neccessary
+        Sets to zero for NonNegativeReals if neccessary
+        """
+        for v_from, v_to in zip(from_list, to_list):
+            if skip_stale and v_from.stale:
+                continue  # Skip stale variable values.
+            if skip_fixed and v_to.is_fixed():
+                continue  # Skip fixed variables.
+            try:
+                v_val = self.get_values(
+                    opt._pyomo_var_to_solver_var_map[v_from])
+                v_to.set_value(v_val)
+                if skip_stale:
+                    v_to.stale = False
+            except ValueError:
+                # Snap the value to the bounds
+                if v_to.lb is not None and v_val < v_to.lb and v_to.lb - v_val <= config.zero_tolerance:
```

Comment:
I would recommend `v_to.has_lb()` in case the default value of `lb` ever changes.